stablehlo/reference/Ops.cpp

```diff
@@ -62,6 +62,25 @@ Tensor evalCeilOp(const Tensor &operand, Type resultType) {
   return result;
 }
 
+Tensor evalConcatenateOp(const ArrayRef<Tensor> &inputs, int64_t dimension,
+                         Type resultType) {
+  Tensor result(resultType);
+  SmallVector<int64_t> offsetIdx(inputs.size());
+  for (size_t offsetDim = 1; offsetDim < inputs.size(); ++offsetDim)
```

Comment:
We can fuse this loop with the following one:

```
Tensor evalConcatenateOp(ArrayRef<Tensor> inputs, int64_t dimension,
                         Type resultType) {
  Tensor result(resultType);

  int64_t offsetDimSize = 0;
  for (size_t inputsDim = 0; inputsDim < inputs.size(); ++inputsDim) {
    for (auto inputIt = inputs[inputsDim].index_begin();
         inputIt != inputs[inputsDim].index_end(); ++inputIt) {
      SmallVector<int64_t> resultIdx(*inputIt);
      resultIdx[dimension] += offsetDimSize;
      result.set(resultIdx, inputs[inputsDim].get(*inputIt));
    }
    offsetDimSize += inputs[inputsDim].getType().getShape()[dimension];
  }
  return result;
}
```

Reply:
Following is the other way round (given a result find the operand index). Either is fine with me. Let us know your opinion. 


```
Tensor evalConcatenateOp(ArrayRef<Tensor> inputs, int64_t dimension,
                         Type resultType) {
  Tensor result(resultType);

  for (auto resultIt = result.index_begin(); resultIt != result.index_end();
       ++resultIt) {
    SmallVector<int64_t> operandIdx(*resultIt);
    for (size_t inputsDim = 0; inputsDim < inputs.size(); ++inputsDim) {
      auto inputDimSize = inputs[inputsDim].getType().getShape()[dimension];
      if (operandIdx[dimension] < inputDimSize) {
        result.set(*resultIt, inputs[inputsDim].get(operandIdx));
        break;
      }
      operandIdx[dimension] -= inputDimSize;
    }
  }

  return result;
}
```

Reply:
Thanks for the suggestion, I'll go with the first option since that avoids a conditional inside the nested loop.